fe/fe-core/src/main/java/org/apache/doris/catalog/RangePartitionInfo.java

```diff
@@ -90,6 +91,8 @@ public PartitionItem createAndCheckPartitionItem(SinglePartitionDesc desc, boole
             if (upperKey.compareTo(newRangeUpper) >= 0) {
                 newRange = checkNewRange(partKeyDesc, newRangeUpper, lastRange, currentRange);
                 break;
+            } else if (partKeyDesc.getPartitionType() == PartitionKeyDesc.PartitionKeyValueType.FIXED) {
```

Comment:
This is not a good algorithm. The original method is optimized for the grammar of `less than`. It will first find a suitable position and then compare it. But for the grammar of `fix range`, it should be judged by comparing the lower value instead of upper value when finding the suitable position.
Therefore, different algorithms should be used for different grammars.